src/transactions/InvokeHostFunctionOpFrame.cpp

```diff
@@ -322,6 +334,463 @@ InvokeHostFunctionOpFrame::maybePopulateDiagnosticEvents(
+                     this](auto const& keys) -> bool {
+        for (auto const& lk : keys)
+        {
+            uint32_t entrySize = 0u;
+            std::optional<TTLEntry> ttlEntry;
+            bool sorobanEntryLive = false;
+
+            // For soroban entries, check if the entry is expired before loading
+            if (isSorobanEntry(lk))
+            {
+                auto ttlKey = getTTLKey(lk);
+                auto ttlIter = entryMap.find(ttlKey);
+                if (ttlIter != entryMap.end() && ttlIter->second.mLedgerEntry)
+                {
+                    if (!isLive(*(ttlIter->second.mLedgerEntry),
+                                ledgerInfo.getLedgerSeq()))
+                    {
+                        // For temporary entries, treat the expired entry as
+                        // if the key did not exist
+                        if (!isTemporaryEntry(lk))
+                        {
+                            if (lk.type() == CONTRACT_CODE)
+                            {
+                                sorobanData.pushApplyTimeDiagnosticError(
+                                    appConfig, SCE_VALUE, SCEC_INVALID_INPUT,
+                                    "trying to access an archived contract "
+                                    "code "
+                                    "entry",
+                                    {makeBytesSCVal(lk.contractCode().hash)});
+                            }
+                            else if (lk.type() == CONTRACT_DATA)
+                            {
+                                sorobanData.pushApplyTimeDiagnosticError(
+                                    appConfig, SCE_VALUE, SCEC_INVALID_INPUT,
+                                    "trying to access an archived contract "
+                                    "data "
+                                    "entry",
+                                    {makeAddressSCVal(
+                                         lk.contractData().contract),
+                                     lk.contractData().key});
+                            }
+                            // Cannot access an archived entry
+                            this->innerResult(res).code(
+                                INVOKE_HOST_FUNCTION_ENTRY_ARCHIVED);
+                            return false;
+                        }
+                    }
+                    else
+                    {
+                        sorobanEntryLive = true;
+                        ttlEntry =
+                            ttlIter->second.mLedgerEntry.value().data.ttl();
+                    }
+                }
+                // If ttlLtxe doesn't exist, this is a new Soroban entry
+            }
+
+            // If ttlLtxe doesn't exist, this is a new Soroban entry
+            // Starting in protocol 23, we must check the Hot Archive for
+            // new keys. If a new key is actually archived, fail the op.
+            if (isPersistentEntry(lk) &&
```

Comment:
I missed the bug fix because my branch was out of date with master, but this block should only run if the ttl entry doesn't exist.